extensions-core/caffeine-cache/src/test/java/io/druid/client/cache/CaffeineCacheTest.java

```diff
@@ -0,0 +1,489 @@
+import java.util.Random;
+import java.util.UUID;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class CaffeineCacheTest
+{
+  private static final byte[] HI = "hiiiiiiiiiiiiiiiiiii".getBytes();
+  private static final byte[] HO = "hooooooooooooooooooo".getBytes();
+
+  private CaffeineCache cache;
+  private final CaffeineCacheConfig cacheConfig = new CaffeineCacheConfig();
+
+  @Before
+  public void setUp() throws Exception
+  {
+    cache = CaffeineCache.create(cacheConfig);
+  }
+
+  @Test
+  public void testBasicInjection() throws Exception
+  {
+    final CaffeineCacheConfig config = new CaffeineCacheConfig();
+    Injector injector = Initialization.makeInjectorWithModules(
+        GuiceInjectors.makeStartupInjector(), ImmutableList.of(
+            new Module()
+            {
+              @Override
+              public void configure(Binder binder)
+              {
+                binder.bindConstant().annotatedWith(Names.named("serviceName")).to("druid/test/redis");
+                binder.bindConstant().annotatedWith(Names.named("servicePort")).to(0);
+
+                binder.bind(CaffeineCacheConfig.class).toInstance(config);
+                binder.bind(Cache.class).toProvider(CaffeineCacheProviderWithConfig.class).in(ManageLifecycle.class);
+              }
+            }
+        )
+    );
+    final Lifecycle lifecycle = injector.getInstance(Lifecycle.class);
+    lifecycle.start();
+    try {
+      Cache cache = injector.getInstance(Cache.class);
+      Assert.assertEquals(CaffeineCache.class, cache.getClass());
+    }
+    finally {
+      lifecycle.stop();
+    }
+  }
+
+  @Test
+  public void testSimpleInjection()
+  {
+    final String uuid = UUID.randomUUID().toString();
+    System.setProperty(uuid + ".type", "caffeine");
+    final Injector injector = Initialization.makeInjectorWithModules(
+        GuiceInjectors.makeStartupInjector(), ImmutableList.<Module>of(
+            new Module()
+            {
+              @Override
+              public void configure(Binder binder)
+              {
+                binder.bindConstant().annotatedWith(Names.named("serviceName")).to("druid/test/redis");
+                binder.bindConstant().annotatedWith(Names.named("servicePort")).to(0);
+
+                binder.bind(Cache.class).toProvider(CacheProvider.class);
+                JsonConfigProvider.bind(binder, uuid, CacheProvider.class);
+              }
+            }
+        )
+    );
+    final CacheProvider cacheProvider = injector.getInstance(CacheProvider.class);
+    Assert.assertNotNull(cacheProvider);
+    Assert.assertEquals(CaffeineCacheProvider.class, cacheProvider.getClass());
+  }
+
+  @Test
+  public void testBaseOps() throws Exception
+  {
+    final Cache.NamedKey aKey = new Cache.NamedKey("a", HI);
+    Assert.assertNull(cache.get(aKey));
+    put(cache, aKey, 1);
+    Assert.assertEquals(1, get(cache, aKey));
+
+    /* Lazily deleted by LRU
```

Comment:
forgot to un-disable a few tests when I added https://github.com/metamx/druid-cache-caffeine/pull/20  fixed